R/functions-Chromatograms.R

```diff
@@ -20,14 +33,37 @@
 #'
 #' @param data A \code{list} of \code{\link{Chromatogram}} objects.
 #'
-#' @param ... Additional parameters to be passed to the \code{\link{matrix}}
-#'     constructor, such as \code{nrow}, \code{ncol} and \code{byrow}.
+#' @param phenoData either a \code{data.frame}, \code{AnnotatedDataFrame} or
+#'     \code{NAnnotatedDataFrame} describing the phenotypical information of the
+#'     samples.
+#'
+#' @param ... Additional parameters to be passed to the
+#'     \code{\link[base]{matrix}} constructor, such as \code{nrow}, \code{ncol}
+#'     and \code{byrow}.
 #' 
 #' @rdname Chromatograms-class
-Chromatograms <- function(data, ...) {
+Chromatograms <- function(data, phenoData, ...) {
     if (missing(data))
         return(new("Chromatograms"))
-    res <- as(matrix(data, ...), "Chromatograms")
+    datmat <- matrix(data, ...)
+    if (missing(phenoData))
+        phenoData <- annotatedDataFrameFrom(datmat, byrow = FALSE)
+    if (ncol(datmat) != nrow(phenoData))
+        stop("Dimensions of the data matrix and the  phenoData do not match")
+    ## Convert phenoData...
+    if (is(phenoData, "data.frame"))
+        phenoData <- AnnotatedDataFrame(phenoData)
+    if (is(phenoData, "AnnotatedDataFrame"))
+        phenoData <- as(phenoData, "NAnnotatedDataFrame")
+    ## Set colnames if we have some
+    if (any(rownames(phenoData) != as.character(1:nrow(phenoData))))
+        colnames(datmat) <- rownames(phenoData)
```

Comment:
Removed now in https://github.com/lgatto/MSnbase/commit/b658d43162c0a964a17709aa8acf64c2c7497d00 as no longer required.